From 073e148f3b2556ef02b331bbf636105733da8480 Mon Sep 17 00:00:00 2001
Message-Id: <073e148f3b2556ef02b331bbf636105733da8480.1670707091.git.lorenzo@kernel.org>
From: Sujuan Chen <sujuan.chen@mediatek.com>
Date: Tue, 29 Nov 2022 11:23:07 +0100
Subject: [PATCH net-next 1/5] wifi: mt76: mt7915: release rxwi in
 mt7915_wed_release_rx_buf

Free rxwi cache releasing WED rx buffers in mt7915_wed_release_rx_buf
routine

Co-developed-by: Lorenzo Bianconi <lorenzo@kernel.org>
Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
Signed-off-by: Sujuan Chen <sujuan.chen@mediatek.com>
---
 dma.c         | 3 ++-
 mt76.h        | 1 +
 mt7915/mmio.c | 2 ++
 3 files changed, 5 insertions(+), 1 deletion(-)

--- a/dma.c
+++ b/dma.c
@@ -165,7 +165,7 @@ mt76_free_pending_txwi(struct mt76_dev *
 	local_bh_enable();
 }
 
-static void
+void
 mt76_free_pending_rxwi(struct mt76_dev *dev)
 {
 	struct mt76_txwi_cache *t;
@@ -178,6 +178,7 @@ mt76_free_pending_rxwi(struct mt76_dev *
 	}
 	local_bh_enable();
 }
+EXPORT_SYMBOL_GPL(mt76_free_pending_rxwi);
 
 static void
 mt76_dma_sync_idx(struct mt76_dev *dev, struct mt76_queue *q)
--- a/mt76.h
+++ b/mt76.h
@@ -1269,6 +1269,7 @@ mt76_tx_status_get_hw(struct mt76_dev *d
 void mt76_put_txwi(struct mt76_dev *dev, struct mt76_txwi_cache *t);
 void mt76_put_rxwi(struct mt76_dev *dev, struct mt76_txwi_cache *t);
 struct mt76_txwi_cache *mt76_get_rxwi(struct mt76_dev *dev);
+void mt76_free_pending_rxwi(struct mt76_dev *dev);
 void mt76_rx_complete(struct mt76_dev *dev, struct sk_buff_head *frames,
 		      struct napi_struct *napi);
 void mt76_rx_poll_complete(struct mt76_dev *dev, enum mt76_rxq_id q,
--- a/mt7915/mmio.c
+++ b/mt7915/mmio.c
@@ -615,6 +615,8 @@ static void mt7915_mmio_wed_release_rx_b
 
 		mt76_put_rxwi(&dev->mt76, t);
 	}
+
+	mt76_free_pending_rxwi(&dev->mt76);
 }
 
 static u32 mt7915_mmio_wed_init_rx_buf(struct mtk_wed_device *wed, int size)
